pallets/xyk/src/lib.rs

```diff
@@ -414,18 +414,20 @@ pub mod pallet {
 
 			let asset_a_reserve = T::Currency::free_balance(asset_a, &pair_account);
 			let asset_b_reserve = T::Currency::free_balance(asset_b, &pair_account);
-			let total_liquidity = Self::total_liquidity(&pair_account);
+			let share_issuance = Self::total_liquidity(&pair_account);
 
-			let amount_b_required =
-				hydra_dx_math::xyk::calculate_liquidity_in(asset_a_reserve, asset_b_reserve, amount_a)
-					.map_err(|_| Error::<T>::AddAssetAmountInvalid)?;
+			let amount_b = hydra_dx_math::xyk::calculate_liquidity_in(asset_a_reserve, asset_b_reserve, amount_a)
+				.map_err(|_| Error::<T>::AddAssetAmountInvalid)?;
 
-			let shares_added = if asset_a < asset_b { amount_a } else { amount_b_required };
+			ensure!(amount_b <= amount_b_max_limit, Error::<T>::AssetAmountExceededLimit);
 
-			ensure!(
-				amount_b_required <= amount_b_max_limit,
-				Error::<T>::AssetAmountExceededLimit
-			);
+			let shares_added = if asset_a < asset_b {
```

Comment:
this if clause is not necessary, calculate_shares(asset_a_reserve, amount_a, share_issuance) and calculate_shares(asset_b_reserve, amount_b, share_issuance) should return the same result (up to rounding error).

Probably best to always use asset a, since amount_b is rounded up and compounding rounding errors could give user too many shares.

Somewhere there should perhaps be a property test that the calculation with assets a & b do in fact return the same result, not sure what exactly the right place for that would be.

Reply:
thanks. changed and tested. Also added one simple test with exact values to make sure that it returns same shares.

i will add a property test for that  too in #501